### PR TITLE
Prepare the VC Data Model v2.0 FPWD publication

### PR DESCRIPTION
**Pull request by msporny**
This PR implements the following changes:

* The ReSpec configuration has been updated to reflect a v2.0 FPWD publication.
* The statements in the Status of the Document section and the changelog have been updated to note that this is a FPWD.
* A FPWD target publication date of 2022-08-04 is proposed.
* A static copy of the FPWD has been created.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/894.html" title="Last updated on Jul 27, 2022, 1:18 PM UTC (f080981)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/894/17329f4...f080981.html" title="Last updated on Jul 27, 2022, 1:18 PM UTC (f080981)">Diff</a>

### CODE REVIEW

File: index.html
Comment (msporny) [inline index.html:95]: @iherman, I need Gabe's `w3cid` value here.

Comment (decentralgabe) [reply index.html:95]: @msporny is it the ID here? `https://www.w3.org/users/116851`?

If so, `116851`.

Comment (msporny) [reply index.html:95]: That's it! Thank you... only you and Ivan can see it. Adding it now.

Comment (msporny) [reply index.html]: Done in 2ffb43dafd9cf7c179fd383e057f77e2eba03fa0.

**Review by decentralgabe — APPROVED** [review]
lgtm

**Comment by iherman** [comment]
Just food for thought... the list of editors is long and that has of course historical reasons. However, it may be better to separate the _current_ editors (who will be responsible for the evolution of the current spec). Wouldn't it be better to have a separate list of "former editors" rather than piling them all up in one place? See, for example, the way we did it for the current EPUB draft spec:

https://www.w3.org/TR/epub-33/

**Comment by msporny** [comment]
> Just food for thought... the list of editors is long and that has of course historical reasons. However, it may be better to separate the _current_ editors (who will be responsible for the evolution of the current spec).

Yes, agreed that the list is getting very long and takes up quite a bit of space and we should think about doing something about it.

I've been thinking about ping'ing the spec-prod team about this and adding a ReSpec rendering feature to render the list of everyone out on a single line.

IMHO, the issue w/ splitting things into "Former Editors" is that they don't show up when you cite the document, and in some cases, the former editors have done far more work on the document than the current Editors -- take a minor revision/maintenance release, for example. 

Fundamentally, ReSpec should probably allow an optional field on which order people should be listed in, where you have people listed in order of contribution across the entire lifetime of the document. We can then do different things w/ the list of "current editors and former editors". 

IOW, if we end up in a situation where a new WG can wipe out the people that did 70% of the work from the credits of a document, when the document is a relatively minor revision of a previous document, we end up in a situation that could be viewed as unethical (from a plagiarism perspective). It also doesn't credit people that have done the work appropriately, leading to a sense of unfairness, and ultimately less contributions as a result.

**Review by OR13 — APPROVED** [review]

**Review by peacekeeper — APPROVED** [review]

**Comment by iherman** [comment]
To be discussed on our meeting later today but, for very practical reasons, I do not think the 4th of August publication date is doable.

The reason is as follows: the publication of the FPWD must be accompanied by a discussion on how the various TR aliases and redirection would take place for things like https://www.w3.org/TR/vc-data-model/, for that plus '/latest' etc. That must be discussed by the WG (which is fine) but, more to the point, must be implemented by the W3C system team. The best person to do that is Denis Ah-Kang, who is on vacations right now, and back on the 8th. 

I would propose we should get all ducks in a row (including the director's decision) for the 11th of August.

**Comment by msporny** [comment]
> I would propose we should get all ducks in a row (including the director's decision) for the 11th of August.

Ok, I'll retarget the FPWD for that date.

**Comment by msporny** [comment]
> I would propose we should get all ducks in a row (including the director's decision) for the 11th of August.

This is now done in https://github.com/w3c/vc-data-model/pull/894/commits/f0809810900ccbe917b7cd9b503cc5e75b28fed7.

**Comment by iherman** [comment]
Having discussed the possible short name setups with @plehegar, here is a proposal for a more general URL set up for the documents on the data model spec.

 We could (ask our system team to) set up the following URL-s/redirections on the `https://www.w3.org/TR/*` space of W3C:

1. Set up a new shortname, namely `vc-data-model-1.1`, referring to the stable 1.1 specification (i.e., the one which, today, is known as https://www.w3.org/TR/vc-data-model).
2. The new `vc-data-model-2.0` will be the short name for the 2.0 version of the data model.
3. The (current) `vc-data-model` would be redirected to `vc-data-model/latest` whose behaviour would be:
   1. to point at the `vc-data-model-1.1` as long as version 2 of the spec is in Working Draft
   2. once 2.0 becomes a CR, it will point at `vc-data-model-2.0`

   (The behaviour is more formally described in [§3.2.1 of the guidebook](https://www.w3.org/2005/05/tr-versions#latest).)

The logic of the third point is that implementers who are not part of the WG want to rely on a stable specification. Today, that is the 1.1 spec. On the other hand, CR means the specification is, technically, stable; i.e., ie, at that point, it is o.k. for implementers to "switch" to 2.0. The net result of these changes is that external developers could continue relying on https://www.w3.org/TR/vc-data-model for the specification and get exactly what they want, and all other details could be considered an internal cuisine of W3C…

cc @plehegar @deniak @caribouW3 to cross check whether my understanding is correct and all this indeed a proper plan to go ahead.

---

P.S. Just as a matter of history, we could also set up a `vc-data-model-1.0` to point at the [1.0 spec](https://www.w3.org/TR/2019/REC-vc-data-model-20191119/). This would not really have a practical impact but would give us a cosy feeling of consistency...

**Comment by deniak** [comment]
Yes, it's probably the best option. Note that the WG can decides the version where https://www.w3.org/TR/vc-data-model should point to, whatever the status of the version is.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-07-27-vcwg#section4-1) on 2022-07-27

List of resolutions:

- ***[Resolution No. 1](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2022-07-27-vcwg#resolution1):  The VCWG will publish the FPWD of VC Data Model v2.0 on 2022-08-11 using the vc-data-model-2.0 shortname and the steps recommended in: [https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080).***

<details><summary><i>View the transcript</i></summary>

#### 4.1. Data model as a FPWD.

_See github pull request [vc-data-model#894](https://github.com/w3c/vc-data-model/pull/894)._




**Ivan Herman:** Not central but... before we go to FPWD is to decide what the short name is.  
… We have to vote for that. What's more interesting is how we handle the various short names vis-a-vis previous version of the data model spec.  
… I had some discussion with PlH about this. I have a proposal for that.  

**Kristina Yasuda:** It seems uncomplicated to have 'v2' in the short name.  

**Ivan Herman:** No, it's not, but we need to ask for it.  

> *Manu Sporny:* See [This is the proposal for shornames:](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080).

**Ivan Herman:** Yes, the VCWG needs to agree what the short URL is, immutable version URLs etc..  
… The background is that we have /TR/vc-data-model.  
… This is the reference for the Rec. The Q is what do we want the future URL to be?.  

> *Manu Sporny:* +1 to the proposal in [https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080) -- it makes sense to me, at least..

> *Manu Sporny:* I'd like us to set up the vc-data-model-1.0 link as well, FWIW..

**Ivan Herman:** It's possible to ask for /TR/vc-data-model-1.1 for the current Rec.  
… we will have /TR/vc-data-model-2.0.  
… We will ask the systems team that the latest stable version... as long as we are in Working Drafts, it will refer to version 1.1.  
… When we get to CR (we're technically done) then the same URl will point to version 2.  
… So the community has a stable reference for a doc that is stable.  

**Phil Archer:** For clarity. The /TR/vc-data-model will imminently resolve to version 1.1 (always to be available at /TR/vc-data-model-1.1).  
… And then will switch to V2 once it gets to CR.  

> *Ivan Herman:* Yes.

> *Charles Lehner:* +1.

**Manu Sporny:** There's a PS. Maybe we should set up a 'version 1.0' URL too..  

**Kristina Yasuda:** Any one against this proposal?.  

**Ivan Herman:** A reminder to ourselves. When we do a formal resolution that we're publishing a FPWD, we have to include the short name that we want.  

> *Phil Archer:* [Discussion about timing cf review time].

**Ivan Herman:** Maybe it's the time to remind the group that any formal resolution only becomes set after 5 days.  

**Brent Zundel:** Makes proposal.  

> *Phil Archer:* [Brent re-writes proposal text].

> **Proposed resolution: use the shortnames and steps recommended in [https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080) and publish the FPWD on 11 Aug 2022.** *(Brent Zundel)*

> **Proposed resolution: The VCWG will publish the FPWD of VC Data Model v2.0 on 2022-08-11 using the vc-data-model-2.0 shortname and the steps recommended in: [https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080).** *(Phil Archer)*

> *Ted Thibodeau Jr.:* +1.

> *Phil Archer:* +1.  

> *Ivan Herman:* +1.

> *Tobias Looker:* +1.

> *Manu Sporny:* +1.

> *Markus Sabadello:* +1.

> *Kevin Dean:* +1.

> *Marty Reed:* +1.

> *Kerri Lemoie:* +1.

> *Gabe Cohen:* +1.

> *Brent Zundel:* +1.

> *Joe Andrieu:* +1.

> *Gregory Natran:* +1.

> *Oliver Terbu:* +1.

> *Steve Cole:* +1.

> *SongPu Ai:* +1.

> ***Resolution #1: The VCWG will publish the FPWD of VC Data Model v2.0 on 2022-08-11 using the vc-data-model-2.0 shortname and the steps recommended in: [https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080](https://github.com/w3c/vc-data-model/pull/894#issuecomment-1196910080).***

**Kristina Yasuda:** We'll leave it to the editors to execute that..  

</details>

**Comment by caribouW3** [comment]
> 1. Set up a new shortname, namely `vc-data-model-1.1`, referring to the stable 1.1 specification (i.e., the one which, today, is known as https://www.w3.org/TR/vc-data-model).
> 2. The new `vc-data-model-2.0` will be the short name for the 2.0 version of the data model.
> 3. The (current) `vc-data-model` would be redirected to `vc-data-model/latest`

The alternative solution would be to point to /TR/vc-data-model/all
and change this when 2.0 reaches a stable enough state (whether it's CR is up to the WG,
but as soon as a WD is used for wide review it might be stable enough)
 
> P.S. Just as a matter of history, we could also set up a `vc-data-model-1.0` to point at the [1.0 spec](https://www.w3.org/TR/2019/REC-vc-data-model-20191119/). This would not really have a practical impact but would give us a cosy feeling of consistency...

It does not appear in /TR/vc-data-model/all, which means that 1.1 is probably considered as
superseding 1.0, in which case you don't want to make it more visible.

**Comment by iherman** [comment]
@caribouW3, the https://www.w3.org/2005/05/tr-versions#shortnames does not list the `/all` suffix for URL-s. First time I see it.

Looking at https://www.w3.org/TR/vc-data-model/all/ today it seems to be an "indirect" link to a table with different versions. Is this its definition? If so, it is indeed useful, but I do not think it is what we are looking for as a URL that non-W3C-savy implementers would be able to use for their own reference.

**Comment by iherman** [comment]
I am still a bit unhappy with the list of editors. This list suggest that the persons are _active contributors_ to the documents that can be contacted by the community on any editorial aspect of the spec. That is no longer the case and, actually, the affiliations are outdated for some on the list (Kyle, Dan, Brent). I do believe that we should find a way of separating the historical editors from active ones in some way or other.

**Comment by iherman** [comment]
Once the merge is done, we should not forget changing the repo's configuration so that the v2.0 branch would be the source of the github pages (as opposed to v1.0 as it is the case right now). I would actually consider renaming 'v2.0' to 'main' (although it is a pain in the neck to do a rename with the number of cloned versions of the repo out there), because it will become a source of bugs if we engage into any kind of github workflow in future (e.g., when we copy an echidna workflow...).

**Review by iherman — APPROVED** [review]

**Comment by msporny** [comment]
Editorial (no normative statements were modified, but this is a significant major version FPWD release), multiple reviews, changes requested and made, no objections, merging.

**Comment by msporny** [comment]
> Once the merge is done, we should not forget changing the repo's configuration so that the v2.0 branch would be the source of the github pages (as opposed to v1.0 as it is the case right now).

This has been merged, and the default branch is now v2.0 AND the pages are served from the v2.0 branch.

> I would actually consider renaming 'v2.0' to 'main' (although it is a pain in the neck to do a rename with the number of cloned versions of the repo out there), because it will become a source of bugs if we engage into any kind of github workflow in future (e.g., when we copy an echidna workflow...).

No need to rename v2.0 to main, Github has the concept of a "default" branch, and that's set to v2.0 right now. All previously checked out repos will continue to work, and Github will suggest that people use the v2.0 branch from now on. We will need to tell ECHIDNA to publish from the v2.0 branch, but that should be it. No disruption necessary. :)

**Pull request merged**
